ui/v2.5/src/components/Tagger/studios/StudioTagger.tsx

```diff
@@ -0,0 +1,750 @@
+        show={showBatchAdd}
+        icon={faStar}
+        header={intl.formatMessage({
+          id: "studio_tagger.add_new_studios",
+        })}
+        accept={{
+          text: intl.formatMessage({
+            id: "studio_tagger.add_new_studios",
+          }),
+          onClick: handleBatchAdd,
+        }}
+        cancel={{
+          text: intl.formatMessage({ id: "actions.cancel" }),
+          variant: "danger",
+          onClick: () => setShowBatchAdd(false),
+        }}
+        disabled={!isIdle}
+      >
+        <Form.Control
+          className="text-input"
+          as="textarea"
+          ref={studioInput}
+          placeholder={intl.formatMessage({
+            id: "studio_tagger.studio_names_separated_by_comma",
+          })}
+          rows={6}
+        />
+        <Form.Text>
+          <FormattedMessage id="studio_tagger.any_names_entered_will_be_queried" />
+        </Form.Text>
+        <div className="mt-2">
+          <Form.Check
+            id="add-parent"
+            checked={batchAddParents}
+            label={intl.formatMessage({
+              id: "studio_tagger.create_or_tag_parent_studios",
+            })}
+            onChange={() => setBatchAddParents(!batchAddParents)}
+          />
+        </div>
+      </ModalComponent>
+      <div className="ml-auto mb-3">
+        <Button onClick={() => setShowBatchAdd(true)}>
+          <FormattedMessage id="studio_tagger.batch_add_studios" />
+        </Button>
+        <Button className="ml-3" onClick={() => setShowBatchUpdate(true)}>
+          <FormattedMessage id="studio_tagger.batch_update_studios" />
+        </Button>
+      </div>
+      <div className={CLASSNAME}>{renderStudios()}</div>
+    </Card>
+  );
+};
+
+interface ITaggerProps {
+  studios: GQL.StudioDataFragment[];
+}
+
+export const StudioTagger: React.FC<ITaggerProps> = ({ studios }) => {
+  const jobsSubscribe = useJobsSubscribe();
+  const intl = useIntl();
+  const { configuration: stashConfig } = React.useContext(ConfigurationContext);
+  const [{ data: config }, setConfig] = useLocalForage<ITaggerConfig>(
+    LOCAL_FORAGE_KEY,
+    initialConfig
+  );
+  const [showConfig, setShowConfig] = useState(false);
+  const [showManual, setShowManual] = useState(false);
+
+  const [batchJobID, setBatchJobID] = useState<string | undefined | null>();
+  const [batchJob, setBatchJob] = useState<JobFragment | undefined>();
+
+  // monitor batch operation
+  useEffect(() => {
+    if (!jobsSubscribe.data) {
+      return;
+    }
+
+    const event = jobsSubscribe.data.jobsSubscribe;
+    if (event.job.id !== batchJobID) {
+      return;
+    }
+
+    if (event.type !== GQL.JobStatusUpdateType.Remove) {
+      setBatchJob(event.job);
+    } else {
+      setBatchJob(undefined);
+      setBatchJobID(undefined);
+
+      // Once the studio batch is complete, refresh all local studio data
+      const ac = getClient();
+      ac.refetchQueries({ include: studioMutationImpactedQueries });
```

Comment:
Again the goal is to refresh the local studio cache, not sure if this is the best way to do it.